### PR TITLE
http.get fail resolver with getUserInterseptor from localForage

### PR DESCRIPTION
**Pull request by VasylPosmit**

### CODE REVIEW

**Comment by VasylPosmit** [comment]
## Reminder:
- [x] To fix _on page and state load_ to only first-time page load

File: src/app/common/interceptors/authentication.js [hunk 1]
```diff
@@ -4,18 +4,12 @@
...
-        request: function(data){
-            console.log('authenticationInterceptor');
-            return $q.resolve(data);
-        },
         responseError: function(rejection) {
           if ($document.find('.ng-dirty').length % 3 === 2) {
-            console.log('authenticationInterceptor');
             if (rejection.status === 401) {
               $window.location.href = 'https://github.com/VasylPosmit/health';//PATH.apps + SIGN_IN_URL;
             }
```
Comment (andrenovax) [inline src/app/common/interceptors/authentication.js:15]: remove it

File: src/app/common/interceptors/getUser.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+(function () {
+  'use strict';
+  angular
+  .module('healthGuide')
+    .factory('getUserInterceptor', getUserInterceptor);
```
Comment (andrenovax) [inline src/app/common/interceptors/getUser.js:5]: why on earth did you make a factory?

File: src/app/common/interceptors/getUser.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+// if the response has error, return data from localForage
+
+  function getUserInterceptor( $q, localForageService ) {
+    /*jshint validthis: true*/
+
+    return {
+        request: function(data){
+            console.log('getUserInterceptor request');
+            return $q.resolve(data);
+        },
```
Comment (andrenovax) [inline src/app/common/interceptors/getUser.js]: you do not need it actually

Comment (andrenovax) [inline src/app/common/interceptors/getUser.js]: think about my comment about cache actually;)

**Comment by andrenovax** [comment]
good-ish;)

File: src/app/common/interceptors/getUser.js (hunk 2 above)
Comment (VasylPosmit) [inline src/app/common/interceptors/getUser.js:5]: service that return instance of itself. Isn't it factory? 
Plus Angular docs [provides example with factory](https://docs.angularjs.org/api/ng/service/$http)
`$provide.factory('myHttpInterceptor',...`

Comment (andrenovax) [inline src/app/common/interceptors/getUser.js:5]: read the example:

```
// alternatively, register the interceptor via an anonymous factory
```

File: src/app/common/interceptors/authentication.js (hunk 1 above)
Comment (VasylPosmit) [inline src/app/common/interceptors/authentication.js:15]: - [x]

File: src/app/common/interceptors/getUser.js (hunk 2 above)
Comment (VasylPosmit) [inline src/app/common/interceptors/getUser.js:5]: I didn't get it.
If I "register the interceptor via an anonymous factory" I can not have it in separate file and insert it into config file like 
`$httpProvider.interceptors.push('getUserInterceptor');
`

> interceptor should be in separate file, consider common/interceprot/authentication.js

**Pull request merged**